apps/organizations/models.py

```diff
@@ -328,6 +329,13 @@ def clean(self):
         if self.twitter_handle and self.twitter_handle[0] == "@":
             self.twitter_handle = self.twitter_handle.replace("@", "")
 
+        self.clean_non_profit()
+
+    def clean_non_profit(self):
+        # Ensure the presence of a Fiscal sponsor automatically treats the RevenueProgram as nonprofit
+        if self.fiscal_sponsor_name is not None and self.fiscal_sponsor_name != "":
```

Comment:
Seems like it would be enough just to test that self.fiscal_sponsor_name is truthy.

```suggestion
        if self.fiscal_sponsor_name:
```